lxd/instance_post.go

```diff
@@ -530,14 +541,9 @@ func instancePostMigration(s *state.State, inst instance.Instance, newName strin
 			return err
 		}
 
-		// If root disk device was not found in target project profiles, apply current root disk device
-		// to the instance's config.
-		if rootDev == nil {
-			rootDevKey, rootDev, err = shared.GetRootDiskDevice(inst.ExpandedDevices().CloneNative())
-			if err != nil {
-				return err
-			}
-
+		// If current root disk device differs from the one found in target project profiles,
+		// set the current one as a local device.
+		if profRootDev == nil || profRootDev["pool"] != rootDev["pool"] {
```

Comment:
It just occurred to me that we potentially also need to consider size and size.state and if they are different also trigger a local device to be used.